src/macros.rs

```diff
@@ -4,6 +4,336 @@ The only public macro is `bitvec`; this module also provides convenience macros
+			&$crate::__bits_store_array!($order, $bits; $($val),*)
+		)
+	};
+	($order:path, $bits:ident; $($val:expr),* $(,)?) => {
+		// Wrap multi-tt `order` into a single opaque tt.
+		$crate::__bits_from_slice!(
+			$order, $bits, $crate::__bits_count!($($val),*),
+			&$crate::__bits_store_array!($order, $bits; $($val),*)
+		)
+	};
+	($order:ident; $($val:expr),* $(,)?) => {
+		$crate::bits!($order, Word; $($val),*)
+	};
+	($order:path; $($val:expr),* $(,)?) => {
+		$crate::bits!($order, Word; $($val),*)
+	};
+	($($val:expr),* $(,)?) => {
+		$crate::bits!(Local, Word; $($val),*)
+	};
+
+	// NOTE: `len` must be `const`, as this is a non-allocating macro.
+	($order:ident, $bits:ident; $val:expr; $len:expr) => {
+		$crate::__bits_from_slice!($order, $bits, $len, &[
+			(0 as $bits).wrapping_sub(($val != 0) as $bits);
+			$crate::store::elts::<$bits>($len)
+		])
+	};
+	($order:path, $bits:ident; $val:expr; $len:expr) => {
+		// Wrap multi-tt `order` into a single opaque tt.
+		$crate::__bits_from_slice!($order, $bits, $len, &[
+			(0 as $bits).wrapping_sub(($val != 0) as $bits);
+			$crate::store::elts::<$bits>($len)
+		])
+	};
+	($order:ident; $val:expr; $len:expr) => {
+		$crate::__bits_from_slice!($order, $crate::store::Word, $len, &[
+			(0 as $crate::store::Word).wrapping_sub(
+				($val != 0) as $crate::store::Word,
+			);
+			$crate::store::elts::<$crate::store::Word>($len)
+		])
+	};
+	($order:path; $val:expr; $len:expr) => {
+		$crate::__bits_from_slice!($order, $crate::store::Word, $len, &[
+			(0 as $crate::store::Word).wrapping_sub(
+				($val != 0) as $crate::store::Word,
+			);
+			$crate::store::elts::<$crate::store::Word>($len)
+		])
+	};
+	($val:expr; $len:expr) => {
+		$crate::__bits_from_slice!(Local, $crate::store::Word, $len, &[
+			(0 as $crate::store::Word).wrapping_sub(
+				($val != 0) as $crate::store::Word,
+			);
+			$crate::store::elts::<$crate::store::Word>($len)
+		])
+	};
+}
+
+// Not public API
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __bits_count {
+	(@ $val:expr) => {
+		1
+	};
+	($($val:expr),*) => {
+		0usize $(+ $crate::__bits_count!(@ $val))*
+	};
+}
+
+// Not public API
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __bits_from_slice {
+	// Ensure literal `Msb0`, `Lsb0`, and `Local` always map to our internal
+	// definitions of the type, as we've been structurally matching them.
+	(Lsb0, $bits:ident, $len:expr, $slice:expr) => {
+		&$crate::slice::BitSlice::<
+			$crate::order::Lsb0,
+			$bits,
+		>::from_slice($slice)[.. $len]
+	};
+	(Msb0, $bits:ident, $len:expr, $slice:expr) => {
+		&$crate::slice::BitSlice::<
+			$crate::order::Msb0,
+			$bits,
+		>::from_slice($slice)[.. $len]
+	};
+	(Local, $bits:ident, $len:expr, $slice:expr) => {
+		&$crate::slice::BitSlice::<
+			$crate::order::Local,
+			$bits,
+		>::from_slice($slice)[.. $len]
+	};
+	(Local, $bits:path, $len:expr, $slice:expr) => {
+		&$crate::slice::BitSlice::<
+			$crate::order::Local,
+			$bits,
+		>::from_slice($slice)[.. $len]
+	};
```

Comment:
This variant can be removed, and all variants can be changed to take `bits` as a `:path`

Reply:
I think this whole *macro* can be removed, since it's no longer delegating to any other macros and just always calling the `from_slice` function…

Reply:
The reason it exists is to ensure that the correct prefix is added to `Local`, `Lsb0`, and `Msb0` orderings.
Without it it would be possible to write `Lsb0`, but have a different order with that name currently in scope, and the macro would do the wrong thing (as it would assume we wanted the internal `Lsb0` ordering, but would build a `BitSlice` with the wrong type parameter).